systemd/systemd.go

```diff
@@ -490,6 +499,48 @@ func (s *systemd) daemonReloadNoLock() error {
 	return err
 }
 
+type ReloadReason struct {
+	UnitsAdded   []string
+	UnitsChanged []string
+}
+
+func (s *systemd) DaemonReloadIfNeeded(reason ReloadReason) error {
+	const locked = false
+	return s.daemonReloadIfNeededWithLock(locked, reason)
+}
+
+func (s *systemd) daemonReloadIfNeededWithLock(locked bool, reason ReloadReason) error {
```

Comment:
nitpick: I would swap the order of the parameters, since the primary action is reloading, and locking is an option.